python/pyarrow/array.pxi

```diff
@@ -2746,6 +2762,29 @@ cdef class StructArray(Array):
         result.validate()
         return result
 
+    def sort(self, order="ascending", fieldname=None):
+        """
+        Sort the StructArray
+
+        Parameters
+        ----------
+        order : "ascending" or "descending"
+            The order of the sorting.
+        fieldname : str or None, default None
+            If to sort the array by one of its fields
+            or by the whole array.
+
+        Returns
+        -------
+        result : StructArray
+        """
+        if fieldname is not None:
+            tosort = self.field(fieldname)
```

Comment:
We currently don't use "fieldname" anywhere. Maybe just `field` as the keyword argument?

Reply:
`field` implied a `Field` instance to me, that's why I went for `fieldname`.